### PR TITLE
Fixes testing problem with spinning channel

### PR DESCRIPTION
**Pull request by pveentjer**

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java [hunk 1]
```diff
@@ -198,7 +199,8 @@ public MemberSocketInterceptor getMemberSocketInterceptor() {
 
     @Override
     public ChannelFactory getChannelFactory() {
-        return new NioChannelFactory();
+        boolean spinning = Boolean.getBoolean("hazelcast.io.spinning");
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java]: The right channel wasn't created.

File: hazelcast/src/main/java/com/hazelcast/internal/networking/spinning/SpinningChannel.java [hunk 2]
```diff
@@ -26,8 +26,8 @@
     private SpinningChannelReader reader;
     private SpinningChannelWriter writer;
 
-    public SpinningChannel(SocketChannel socketChannel, boolean client) {
-        super(socketChannel, client);
+    public SpinningChannel(SocketChannel socketChannel, boolean clientMode) {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/networking/spinning/SpinningChannel.java]: We use clientMode on other places. So immediately aligned the name

File: hazelcast/src/main/java/com/hazelcast/internal/networking/spinning/SpinningChannelFactory.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package com.hazelcast.internal.networking.spinning;
+
+import com.hazelcast.internal.networking.Channel;
+import com.hazelcast.internal.networking.ChannelFactory;
+
+import java.nio.channels.SocketChannel;
+
+public class SpinningChannelFactory implements ChannelFactory {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/networking/spinning/SpinningChannelFactory.java]: This was the problem; no SpinningChannelFactory; so all channels created were NioChannel.

File: hazelcast/src/main/java/com/hazelcast/internal/networking/spinning/SpinningChannelReader.java [hunk 4]
```diff
@@ -111,6 +109,13 @@ private boolean init() throws IOException {
         }
         this.inboundHandler = init.getHandler();
         this.inputBuffer = init.getByteBuffer();
+
+        if (inboundHandler instanceof ChannelInboundHandlerWithCounters) {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/networking/spinning/SpinningChannelReader.java]: Another problem; counters were not initialized.

File: hazelcast/src/test/java/com/hazelcast/nio/tcp/TcpIpConnection_AbstractTest.java [hunk 5]
```diff
@@ -40,7 +40,7 @@
 
 public abstract class TcpIpConnection_AbstractTest extends HazelcastTestSupport {
 
-    protected EventLoopGroupFactory threadingModelFactory = new Select_NioEventLoopGroupFactory();
+    protected EventLoopGroupFactory eventLoopGroupFactory = new Select_NioEventLoopGroupFactory();
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/nio/tcp/TcpIpConnection_AbstractTest.java]: old name

**Comment by jerrinot** [comment]
run-lab-run

**Pull request merged**